web3.md

```diff
@@ -1040,6 +1042,42 @@ Transaction Receipts
 
     rule <k> #eth_estimateGas_finalize _ => #popNetworkState ~> #sendResponse ( "error": {"code": -32000, "message":"base fee exceeds gas limit"}) ... </k>
          <statusCode> EVMC_OUT_OF_GAS </statusCode>
+```
+
+Collecting Coverage Data
+------------------------
 
+- `<execPhase>` cell is used to differentiate between the generated code used for contract deployment and the bytecode of the contract.
+- `<opcodeCoverage>` cell is a map which stores the program counters which were hit during the execution of a program. The key, named `CoverageIdentifier`, contains the hash of the bytecode which is executed, and the phase of the execution.
+
+```k
+    syntax Phase ::= ".Phase"
+                   | "CONSTRUCTOR"
+                   | "RUNTIME"
+
+    syntax CoverageIdentifier ::= "{" Int "|" Phase "}"
+
+    rule <k> #mkCall _ _ _ _ _ _ _ ... </k>
+         <execPhase> ( EPHASE => RUNTIME ) </execPhase>
+      requires EPHASE =/=K RUNTIME
+      [priority(25)]
+
+    rule <k> #mkCreate _ _ _ _ ... </k>
+         <execPhase> ( EPHASE => CONSTRUCTOR ) </execPhase>
+      requires EPHASE =/=K CONSTRUCTOR
+      [priority(25)]
+
+    rule <k> #initVM ... </k>
+         <opcodeCoverage> OC => OC [ ListItem({keccak(CODE) | EPHASE}) <- .Set ] </opcodeCoverage>
+         <execPhase> EPHASE </execPhase>
+         <program> CODE </program>
+      requires notBool ListItem({keccak(CODE) | EPHASE}) in_keys(OC)
+      [priority(25)]
+
+    rule <k> #execute ... </k>
+         <pc> PCOUNT </pc>
+         <opcodeCoverage> OC[ ListItem({keccak(CODE) | EPHASE}) |-> ( (PCS .Set) => SetItem(PCOUNT) ) ] </pcVisited>
```

Comment:
This is dropping all the existing `PCS` each time.

Should probably be more like `... { keccak(CODE) | EPHASE } |-> (PCS (.Set => SetItem(PCOUNT))) ...`